src/treetensornetworks/solvers/alternating_update.jl

```diff
@@ -1,62 +1,17 @@
 
-function _extend_sweeps_param(param, nsweeps)
-  if param isa Number
-    eparam = fill(param, nsweeps)
-  else
-    length(param) >= nsweeps && return param[1:nsweeps]
-    eparam = Vector(undef, nsweeps)
-    eparam[1:length(param)] = param
-    eparam[(length(param) + 1):end] .= param[end]
-  end
-  return eparam
-end
-
-function process_sweeps(
-  nsweeps;
-  cutoff=fill(1E-16, nsweeps),
-  maxdim=fill(typemax(Int), nsweeps),
-  mindim=fill(1, nsweeps),
-  noise=fill(0.0, nsweeps),
-  kwargs...,
-)
-  maxdim = _extend_sweeps_param(maxdim, nsweeps)
-  mindim = _extend_sweeps_param(mindim, nsweeps)
-  cutoff = _extend_sweeps_param(cutoff, nsweeps)
-  noise = _extend_sweeps_param(noise, nsweeps)
-  return maxdim, mindim, cutoff, noise, kwargs
-end
-
-function sweep_printer(; outputlevel, state, which_sweep, sw_time)
-  if outputlevel >= 1
-    print("After sweep ", which_sweep, ":")
-    print(" maxlinkdim=", maxlinkdim(state))
-    print(" cpu_time=", round(sw_time; digits=3))
-    println()
-    flush(stdout)
-  end
-end
-
 function alternating_update(
-  updater,
   projected_operator,
   init_state::AbstractTTN;
-  checkdone=(; kws...) -> false,
-  outputlevel::Integer=0,
-  nsweeps::Integer=1,
+  sweep_plans,  #this is really the only one beig pass all the way down
+  outputlevel, # we probably want to extract this one indeed for passing to observer etc.
+  checkdone=(; kws...) -> false,  ### move outside
   (sweep_observer!)=observer(),
   sweep_printer=sweep_printer,
-  write_when_maxdim_exceeds::Union{Int,Nothing}=nothing,
-  updater_kwargs,
-  kwargs...,
+  write_when_maxdim_exceeds::Union{Int,Nothing}=nothing, ### move outside
 )
-  maxdim, mindim, cutoff, noise, kwargs = process_sweeps(nsweeps; kwargs...)
-
   state = copy(init_state)
-
-  insert_function!(sweep_observer!, "sweep_printer" => sweep_printer) # FIX THIS
-
-  for which_sweep in 1:nsweeps
-    if !isnothing(write_when_maxdim_exceeds) &&
+  for (which_sweep, sweep_plan) in enumerate(sweep_plans)
+    if !isnothing(write_when_maxdim_exceeds) && #fix passing this
       maxdim[which_sweep] > write_when_maxdim_exceeds
       if outputlevel >= 2
         println(
```

Comment:
This entire logic around `write_when_maxdim_exceeds` could be handled through a callback function that gets passed down from the top level. For example, we could pass down a function like:
```julia
function cache_to_disk(operator; which_sweep, maxdim, outputlevel, write_when_maxdim_exceeds)
    maxdim[which_sweep] > write_when_maxdim_exceeds
    if outputlevel >= 2
      println(
        "write_when_maxdim_exceeds = $write_when_maxdim_exceeds and maxdim[which_sweep] = $(maxdim[which_sweep]), writing environment tensors to disk",
      )
    end
    operator = disk(operator)
  end
  return operator
end
```
as a keyword argument `transform_operator` passed down from `alternating_update`, `tdvp`, etc. `write_when_maxdim_exceeds` could get passed down from a `transform_operator_kwargs` named tuple, also passed down from `alternating_update`, `tdvp`, etc.

Then, this could would just be replaced by a call to `transform_operator(projected_operator; which_sweep, maxdim, outputlevel, transform_operator_kwargs...)`, which is simpler for this code and more general.

Reply:
Are you suggesting to put `transform_operator_kwargs` in the sweep_plan, like `updater_kwargs` etc? Or to treat it separately and pass it to `alternating_update` only? The former would move it from `alternating_update` to `update_region`, not sure if that's a problem in terms of repeating the call to `disk`?

Reply:
That's a good question. Probably ultimately it does make sense for it to be processed as part of the sweep plan, but maybe we can leave this change for future work since there are some nontrivial choices to make.